### DISCUSSION TITLE
Creating a new plugin doesn't seem to work

### DISCUSSION THREAD

**Discussion by utkarshk384** [tailwindlabs/tailwindcss#5953] [Help]
**What version of Tailwind CSS are you using?**

2.2.17

**What build tool (or framework if it abstracts the build tool) are you using?**

svelte 3.37.0, vite 2.6.13 typescript 4.3.2

**What version of Node.js are you using?**

v 14.17.0

**What browser are you using?**

Firefox

**What operating system are you using?**

macOS

**Reproduction URL**

https://github.com/utkarshk384/repo-tailwindcss

**Describe your issue**

So I have been trying to create a new plugin for tailwind using typescript. However, whenever I compile typescript files to js and import the plugin in `tailwind.config.cjs`, I receive an error that says `[vite] Internal server error: plugin is not a function` when using `exports.default` but it gets fixed if I use `module.exports`   Here is my `tsconfig.json` that was used to compile to js:

```json
{
    "compilerOptions": {
        "noEmit": false,
        "rootDir": "src",
        "declaration": true,
        "declarationDir": "dist/types",
        "module": "CommonJS",
        "outDir": "dist/"
    },
    "include": ["src"]
}
```
## Steps to repo
1. git clone [this repo](https://github.com/utkarshk384/repo-tailwindcss) to your local environment.
2. Then run install all the dependencies including the custom `myplugin` that is in the directory.
3. Then navigate to `./myPlugin/dist/index.js` and experiment with the two export types.
4. I also created a handy command that would reinstall my custom plugin. The command is `yarn refresh`

Any help is appreciated

**Comment by N00nDay** [comment]
Have you figured out a solution to this issue? I am dealing with the same problem.
